knowledge_repo/app/routes/posts.py

```diff
@@ -172,3 +177,33 @@ def render_legacy():
 def about():
     """Renders about page. This is the html version of REAMDE.md"""
     return render_template("about.html")
+
+
+@blueprint.route('/ajax/post/download', methods=['GET'])
+@PageView.logged
+def download():
+    "Downloads resources associated with a post."
+
+    path = request.args.get('post')
```

Comment:
I'm ambivalent here. I added it to indicate to someone unfamiliar with the code that the appropriate way to identify a post is by path; if that isn't already obvious.